connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTask.java

```diff
@@ -321,7 +321,7 @@ private void commitTransaction() {
 
         error = flushError.get();
         if (error != null) {
-            recordCommitFailure(time.milliseconds() - started, null);
+            recordCommitFailure(time.milliseconds() - started);
```

Comment:
This changes the behavior of `recordCommitFailure()`. When calling the method with 2 arguments, it then calls `recordCommit()` with `success` set to `false` which in turn impacts metrics.

Reply:
The `recordCommit()` method doesn't use the `Throwable error` argument, so I removed it.
The behavior of `recordCommitFailure()` remains the same.

Reply:
Yes but the `success` field has an impact.

Reply:
I'm sorry I don't understand what the behavioral change is.

In trunk:
```
protected void recordCommitFailure(long duration, Throwable error) {
    taskMetricsGroup.recordCommit(duration, false, error);
 }

void recordCommit(long duration, boolean success, Throwable error) {
    if (success) {
        commitTime.record(duration);
        commitAttempts.record(1.0d);
     } else {
        commitAttempts.record(0.0d);
     }
 }
```

In current brunch:
```
protected void recordCommitFailure(long duration) {
      taskMetricsGroup.recordCommit(duration, false);
}

void recordCommit(long duration, boolean success) {
    if (success) {
       commitTime.record(duration);
       commitAttempts.record(1.0d);
    } else {
        commitAttempts.record(0.0d);
    }
}
```

I only removed the `Throwable error` argument since it was not used inside the `recordCommit()` method.

Reply:
My bad, I only took a quick look and got confused with `recordCommitSuccess()`. You're right the `Throwable` argument can be removed.